### PR TITLE
FIx dataset collator for LLaMA finetuning

### PR DESCRIPTION
**Pull request by gongel**
### PR types

Bug fixes
### PR changes

Others
### Description

- Add virtual_pp_degree for pipeline training
- Add attention mask when `eval_with_do_generation=True`
- Labels need to be padded to max_length

### CODE REVIEW

File: examples/language_model/llama/modeling_pp.py [hunk 1]
```diff
@@ -218,7 +218,8 @@ def __init__(
 
         use_recompute = self.config.use_recompute
         recompute_granularity = self.config.recompute_granularity
-        virtual_pp_degree = self.config.virtual_pp_degree
+        # virtual_pp_degree = self.config.virtual_pp_degree
+        virtual_pp_degree = getattr(self.config, "virtual_pp_degree", 1)
```
Comment (ZHUI) [inline examples/language_model/llama/modeling_pp.py:222]: LG for virtual_pp_degree

**Review by ZHUI — APPROVED** [review]

**Review by w5688414 — APPROVED** [review]
LGTM

**Review by sijunhe — APPROVED** [review]
